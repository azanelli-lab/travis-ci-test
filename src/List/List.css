.c-list {
  padding-left: 0;
}

.c-list__item {
  list-style-type: none;
  /* background-color: gray; */
  padding: 20px 10px;
  margin: 10px 10px;
  border: 2px solid #444;
  font-size: 1.2rem;
}

.c-list__item:hover {
  border: 2px solid #333;
  /* background-color: #ccc; */
  box-shadow: 0px 0px 10px 1px #ddd;
}
