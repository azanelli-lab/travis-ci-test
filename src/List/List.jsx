import React from 'react';
import PropTypes from 'prop-types';

class List extends React.Component {

  static propTypes = {
    items: PropTypes.array.isRequired,
  }

  renderList = () => {
    const renderedList = this.props.items.map((item) => {
      return (<li key={item.id} className="c-list__item">{item.label}</li>);
    });

    return renderedList;
  }

  render() {
    return (
      <ul className="c-list">
        { this.renderList() }
      </ul>
    );
  }
}

export default List;
