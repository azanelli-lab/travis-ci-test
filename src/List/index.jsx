import List from './List.jsx';
import './List.css';
export default List;
