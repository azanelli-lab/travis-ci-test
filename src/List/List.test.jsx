import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('<List />', () => {

  const items = [
    { id: 1, label: 'test 1' },
    { id: 2, label: 'test 2' },
  ];

  it('renders without crashing', () => {
    const component = shallow(<List items={items} />);
    expect(component).toMatchSnapshot();
  });

  it('renders all the items', () => {
    const component = shallow(<List items={items} />);
    expect(component.find('.c-list__item').length).toBe(2);
  });

});
