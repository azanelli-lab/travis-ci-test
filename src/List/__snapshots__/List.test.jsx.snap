// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={
        Array [
            Object {
              "id": 1,
              "label": "test 1",
            },
            Object {
              "id": 2,
              "label": "test 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="c-list__item"
>
          test 1
</li>,
        <li
          className="c-list__item"
>
          test 2
</li>,
      ],
      "className": "c-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "test 1",
          "className": "c-list__item",
        },
        "ref": null,
        "rendered": "test 1",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "test 2",
          "className": "c-list__item",
        },
        "ref": null,
        "rendered": "test 2",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="c-list__item"
>
            test 1
</li>,
          <li
            className="c-list__item"
>
            test 2
</li>,
        ],
        "className": "c-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "test 1",
            "className": "c-list__item",
          },
          "ref": null,
          "rendered": "test 1",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "test 2",
            "className": "c-list__item",
          },
          "ref": null,
          "rendered": "test 2",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
